src/persist-client/src/internal/paths.rs

```diff
@@ -49,6 +51,60 @@ impl PartId {
     }
 }
 
+/// A component that provides information about the writer of a blob.
+/// For older blobs, this is a UUID for the specific writer instance;
+/// for newer blobs, this is a string representing the version at which the blob was written.
+/// In either case, it's used to help determine whether a blob may eventually
+/// be linked into state, or whether it's junk that we can clean up.
+#[derive(Debug, Ord, PartialOrd, Eq, PartialEq, Clone)]
+pub enum WriterKey {
+    Id(WriterId),
```

Comment:
the minimum-version stuff makes the ordering of variants in this enum structural, correct? I think that deserves a comment